### DISCUSSION TITLE
DIY BMS V4

### DISCUSSION THREAD

**Discussion by dakoal** [Louisvdw/dbus-serialbattery#554] [New BMS requests]
https://github.com/stuartpittaway/diyBMSv4

This BMS would be fine to be integrated

**Comment by Louisvdw** [comment]
Hi @dakoal 
Is there a communication protocol document available?
I can see there is an MQTT and NodeRed server, but is there a serial interface?

**Comment by dakoal** [comment]
I am in contact with Stuart. Maybe in the near future the MQTT JSON output will be delivered on the USB/Serial Port.
When you connect a DIY BMS on the USB Port of a cerbo, you got the data every second (now its only debug data, but its coming on /dev/ttyUSBXY).

I will inform here when it's available, provide Test data and the format.

https://github.com/stuartpittaway/diyBMSv4ESP32/issues/22

**Comment by Louisvdw** [comment]
That would be perfect. It it connects to /dev/ttyUSB then the serial-started service on the Venus can start my driver. If we can read the battery data from there it will work.

**Comment by Louisvdw** [comment]
As a minimum we need to know: 
The amount of cells in the battery 
And then also the SOC, Voltage, Current.

Cell data, temp data, alarms, charge limits are all extras

**Comment by CuriousTimo** [comment]
Stuart has implemented CanBus on the diyBMSv4. This is directly compatible with Victron.
So I see no need anymore to connect the diyBMSv4 over serial.

**Comment by dakoal** [comment]
That is true when using a cerbo.

But VenusOS on a raspi doesnt have canbus.

Also there are many „old“ diybms Controller PCB‘s out there having no CANBUS

**Comment by CuriousTimo** [comment]
@dakoal Just add an MCP2515 CAN-Bus HAT.
https://github.com/victronenergy/venus/wiki/RaspberryPi-CAN-Interfaces
